packages/backend/src/database/migrations/20220428071116_job_steps.ts

```diff
@@ -32,7 +32,8 @@ export async function up(knex: Knex): Promise<void> {
                 .defaultTo(knex.fn.now());
             tableBuilder.text('step_status').notNullable();
             tableBuilder.text('step_type').notNullable();
-            tableBuilder.text('step_label').notNullable();
+            tableBuilder.text('step_error');
+            tableBuilder.unique(['job_uuid', 'step_type']);
```

Comment:
Hot dang!!!! We got a constraint!

This will also setup an index on this pair so makes lookups much faster 😀